src/main/java/spoon/support/reflect/declaration/CtElementImpl.java

```diff
@@ -186,6 +190,24 @@ public String getDocComment() {
 		return "";
 	}
 
+	public CtCompilationUnit getOrCreateCompilationUnit() {
+		CompilationUnitFactory cuFactory = getFactory().CompilationUnit();
+		if (this instanceof CtType<?>) {
+			return cuFactory.getOrCreate((CtType<?>) this);
+		} else if (this instanceof CtPackage) {
+			return cuFactory.getOrCreate((CtPackage) this);
+		} else if (this instanceof CtModule) {
+			return cuFactory.getOrCreate((CtModule) this);
+		} else {
+			CtType<?> enclosingType = getParent(CtType.class);
+			if (enclosingType != null) {
+				return cuFactory.getOrCreate(enclosingType);
+			} else {
+				throw new SpoonException(this.getShortRepresentation() + " does not belong to a compilation unit");
```

Comment:
Does using `this.getShortRepresentation()` better the error message in some way? If not, I am willing to replace it with "The element".